subsystem/user_social/get_event_marathon_ranking_user.go

```diff
@@ -2,12 +2,28 @@ package user_social
 
 import (
 	"elichika/client"
+	"elichika/subsystem/cache"
 	"elichika/userdata"
 	"elichika/utils"
 )
 
+var (
+	getEventMarathonRankingUserCache = cache.UniquePointerMap[int64, cache.CachedObject[client.EventMarathonRankingUser]]{}
+)
+
 func GetEventMarathonRankingUser(session *userdata.Session, userId int32) client.EventMarathonRankingUser {
-	// TODO(cache): Maybe cache this
+	key := (int64(userId) << 32)
```

Comment:
Is there a need to shift `userId` up (or use int64 map at all)?

I think it's fine to use map from int32 and use use userId as the key.